### ISSUE TITLE
Adic7ed provider failed with x-files series

### ISSUE THREAD

**Issue by mac1202** [Diaoul/subliminal#563]
Subliminal failed to find subtitle for the new x-files episodes. Here the error i got

    INFO:subliminal.api:Listing subtitles with provider 'addic7ed' and languages {<Language [fr]>}
    INFO:subliminal.api:Initializing provider addic7ed
    INFO:subliminal.providers.addic7ed:Logging in
    DEBUG:subliminal.providers.addic7ed:Logged in
    ERROR:subliminal.api:Unexpected error in provider 'addic7ed', discarding it
    Traceback (most recent call last):
      File "/usr/lib/python3.5/site-packages/subliminal/api.py", line 116, in list_subtitles
        provider_subtitles = self[name].list_subtitles(video, provider_languages)
      File "/usr/lib/python3.5/site-packages/subliminal/providers/addic7ed.py", line 259, in list_subtitles
        return [s for s in self.query(video.series, video.season, video.year)
      File "/usr/lib/python3.5/site-packages/subliminal/providers/addic7ed.py", line 214, in query
        show_id = self.get_show_id(series, year, country)
      File "/usr/lib/python3.5/site-packages/subliminal/providers/addic7ed.py", line 187, in get_show_id
        show_ids = self._get_show_ids()
      File "/usr/lib/python3.5/site-packages/dogpile/cache/region.py", line 1053, in decorate
        should_cache_fn)
      File "/usr/lib/python3.5/site-packages/dogpile/cache/region.py", line 657, in get_or_create
        async_creator) as value:
      File "/usr/lib/python3.5/site-packages/dogpile/core/dogpile.py", line 158, in __enter__
        return self._enter()
      File "/usr/lib/python3.5/site-packages/dogpile/core/dogpile.py", line 91, in _enter
        value = value_fn()
      File "/usr/lib/python3.5/site-packages/dogpile/cache/region.py", line 610, in get_value
        value = self.backend.get(key)
      File "/usr/lib/python3.5/site-packages/dogpile/cache/backends/file.py", line 220, in get
        with self._dbm_file(False) as dbm:
      File "/usr/lib/python3.5/contextlib.py", line 59, in __enter__
        return next(self.gen)
      File "/usr/lib/python3.5/site-packages/dogpile/cache/backends/file.py", line 215, in _dbm_file
        "w" if write else "r")
      File "/usr/lib/python3.5/dbm/__init__.py", line 88, in open
        raise error[0]("db type could not be determined")
    dbm.error: db type could not be determined

**Comment by Diaoul** [comment]
It seems like a cache issue. Please remove the cache file in ~/.config/subliminal/ and try again.

**Comment by mac1202** [comment]
ok got a new error message now 

    INFO:subliminal.api:Initializing provider addic7ed
    INFO:subliminal.providers.addic7ed:Logging in
    DEBUG:subliminal.providers.addic7ed:Logged in
    DEBUG:subliminal.providers.addic7ed:Getting show id
    INFO:subliminal.providers.addic7ed:Getting the page of show id 85, season 10
    ERROR:subliminal.api:Unexpected error in provider 'addic7ed', discarding it
    Traceback (most recent call last):
      File "/usr/lib/python3.5/site-packages/subliminal/api.py", line 116, in list_subtitles
        provider_subtitles = self[name].list_subtitles(video, provider_languages)
      File "/usr/lib/python3.5/site-packages/subliminal/providers/addic7ed.py", line 259, in list_subtitles
        return [s for s in self.query(video.series, video.season, video.year)
      File "/usr/lib/python3.5/site-packages/subliminal/providers/addic7ed.py", line 229, in query
        series = match.group('series')
    AttributeError: 'NoneType' object has no attribute 'group'

**Comment by mac1202** [comment]
my mistake my folder permission wasnt properly set.
now i got a new error 

    INFO:subliminal.api:Listing subtitles with provider 'addic7ed' and languages {<Language [fr]>}
    INFO:subliminal.api:Initializing provider addic7ed
    INFO:subliminal.providers.addic7ed:Logging in
    DEBUG:subliminal.providers.addic7ed:Logged in
    DEBUG:subliminal.providers.addic7ed:Getting show id
    INFO:subliminal.providers.addic7ed:Getting the page of show id 85, season 10
    ERROR:subliminal.api:Unexpected error in provider 'addic7ed', discarding it
    Traceback (most recent call last):
      File "/usr/lib/python3.5/site-packages/subliminal/api.py", line 116, in list_subtitles
        provider_subtitles = self[name].list_subtitles(video, provider_languages)
      File "/usr/lib/python3.5/site-packages/subliminal/providers/addic7ed.py", line 259, in list_subtitles
        return [s for s in self.query(video.series, video.season, video.year)
      File "/usr/lib/python3.5/site-packages/subliminal/providers/addic7ed.py", line 224, in query
        raise TooManyRequests()
    subliminal.exceptions.TooManyRequests

**Comment by Diaoul** [comment]
Well, you made too many requests to addic7ed and you're now limited. Wait a few hours and try again.

**Comment by mac1202** [comment]
ok i wait and too many request error is gone, but the previous one "AttributeError: 'NoneType' object has no attribute 'group'" is still here.

**Comment by kakai248** [comment]
Please check my pull request. It fixes the problem with X-Files and eventually other series that have a dash in the name.

**Comment by mac1202** [comment]
kakai248 I compiled your hotfix branch, but addic7ed provider failed to login with it.

    INFO:subliminal.api:Listing subtitles with provider 'addic7ed' and languages {<Language [fr]>}
    INFO:subliminal.api:Initializing provider addic7ed
    INFO:subliminal.providers.addic7ed:Logging in
    ERROR:subliminal.api:Unexpected error in provider 'addic7ed', discarding it
    Traceback (most recent call last):
      File "/usr/lib/python3.5/site-packages/subliminal/api.py", line 116, in list_subtitles
        provider_subtitles = self[name].list_subtitles(video, provider_languages)
      File "/usr/lib/python3.5/site-packages/subliminal/api.py", line 61, in __getitem__
        provider.initialize()
      File "/usr/lib/python3.5/site-packages/subliminal/providers/addic7ed.py", line 100, in initialize
        raise AuthenticationError(self.username)
    subliminal.exceptions.AuthenticationError: mac1202

**Comment by kakai248** [comment]
That is weird. The only thing I changed was adding a dash to this line:
```
series_year_re = re.compile('^(?P<series>[ \w\'.:-]+)(?: \((?P<year>\d{4})\))?$')
```
It doesn't even affect the authentication phase.

**Comment by mac1202** [comment]
I just retry downloading subtitle and this time it's work strange :) 
So I confirm your PR is fixing this issue.

**Comment by Diaoul** [comment]
Fixed with d125ffa
